module/evdi_drv.c

```diff
@@ -61,7 +61,9 @@ static const struct file_operations evdi_driver_fops = {
 
 static struct drm_driver driver = {
 	.driver_features = DRIVER_MODESET | DRIVER_GEM | DRIVER_PRIME,
+#if KERNEL_VERSION(4, 12, 0) <= LINUX_VERSION_CODE
```

Comment:
Sorry, I some of these swapped around - they're really fiddly! I will post a revised version addressing these and your other comments.